proposals/0000-exportable-named-default.rst

```diff
@@ -0,0 +1,406 @@
+default, they just have to declare it in the importing module. Furthermore, if they export this ``default``
+declaration, every importer of the module will have the conflicts resolved for them::
+
+   module ProjectImports (Text.Text, Foundation.String,
+                          default IsString)
+
+   import qualified Data.Text         as Text
+   import qualified Foundation.String as Foundation
+
+   default IsString (Text.Text, Foundation.String, String)
+
+An equivalent story can be told for the ``OverloadedLists``, by replacing ``Text`` and ``Foundation.String`` by
+``Vector`` and ``Foundation.String`` by ``Foundation.Array``.
+
+Effect and Interactions
+-----------------------
+
+GHC already supports two extensions that modify the defaulting mechanism:
+`ExtendedDefaultRules <https://ghc.gitlab.haskell.org/ghc/doc/users_guide/ghci.html#type-defaulting-in-ghci>`_ and
+`OverloadedStrings <https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/overloaded_strings.html?highlight=overloadedstrings#overloaded-string-literals>`_.
+
+ExtendedDefaultRules
+~~~~~~~~~~~~~~~~~~~~
+
+The former is fully devoted to defaulting. Its effect is to extend the defaulting rules so that they apply not only to
+the class ``Num`` as specified by the language standard, but also to any class in the following list: ``Show``,
+``Eq``, ``Ord``, ``Foldable``, ``Traversable``, or any numeric class. This list is hard-coded and not
+user-extensible. Furthermore, the extension adds ``()`` and ``[]`` to the list of default types to try. If the present
+proposal is accepted, ``ExtendedDefaultRules`` could be reformulated as a set of actual ``default`` declarations
+brought into the scope::
+
+  default Show ((), Integer, Double)
+  default Eq ((), Integer, Double)
+  default Ord ((), Integer, Double)
+  default Foldable ([])
+  default Traversable ([])
+  default Num ((), Integer, Double)
+
+OverloadedStrings
+~~~~~~~~~~~~~~~~~
+
+The ``OverloadedStrings`` extension by itself causes many new ambiguities, much like the ambiguites caused by the
+overloaded numeric literals which were the original reason for ``default`` declarations in the first place. To rectify
+this problem, the extension tweaks the defaulting mechanism. To quote from the GHC manual:
+
+- Each type in a ``default`` declaration must be an instance of ``Num`` or of ``IsString``.
+
+- If no ``default`` declaration is given, then it is just as if the module contained the declaration ``default
+  (Integer, Double, String)``.
+
+- The standard defaulting rule is extended thus: defaulting applies when all the unresolved constraints involve
+  standard classes or ``IsString``; and at least one is a numeric class or ``IsString``.
+
+Once again, if the present proposal were adopted, the above rules could be expressed as an actual ``default``
+declaration::
+
+   default IsString (Integer, Double, String)
```

Comment:
Doesn’t this imply that `"x"` would be defaulted to `"x" :: Integer`? How does that make any sense?